spec/lib/mastodon/cli/accounts_spec.rb

```diff
@@ -1361,4 +1361,183 @@
+      allow(MoveService).to receive(:new).and_return(move_service)
+    end
+
+    shared_examples 'a successful migration' do
+      it 'calls the MoveService for the last migration' do
+        cli.invoke(:migrate, arguments, options)
+
+        last_migration = source_account.migrations.last
+
+        expect(move_service).to have_received(:call).with(last_migration).once
+      end
+
+      it 'displays a successful message' do
+        expect { cli.invoke(:migrate, arguments, options) }.to output(
+          a_string_including("OK, migrated #{source_account.acct} to #{target_account.acct}")
+        ).to_stdout
+      end
+    end
+
+    context 'when both --replay and --target options are given' do
+      let(:options) { { replay: true, target: "#{target_account.username}@example.com" } }
+
+      it 'exits with an error message indicating that using both options is not possible' do
+        expect { cli.invoke(:migrate, arguments, options) }.to output(
+          a_string_including('Use --replay or --target, not both')
+        ).to_stdout
+          .and raise_error(SystemExit)
+      end
+    end
+
+    context 'when no option is given' do
+      it 'exits with an error message indicating that at least one option must be used' do
+        expect { cli.invoke(:migrate, arguments, {}) }.to output(
+          a_string_including('Use either --replay or --target')
+        ).to_stdout
+          .and raise_error(SystemExit)
+      end
+    end
+
+    context 'when the given username is not found' do
+      let(:arguments) { ['non_existent_username'] }
+
+      it 'exits with an error message indicating that there is no such account' do
+        expect { cli.invoke(:migrate, arguments, replay: true) }.to output(
+          a_string_including("No such account: #{arguments.first}")
+        ).to_stdout
+          .and raise_error(SystemExit)
+      end
+    end
+
+    context 'with --replay option' do
+      let(:options) { { replay: true } }
+
+      context 'when the specified account has no previous migrations' do
+        it 'exits with an error message indicating that the given account has no previous migrations' do
+          expect { cli.invoke(:migrate, arguments, options) }.to output(
+            a_string_including('The specified account has not performed any migration')
+          ).to_stdout
+            .and raise_error(SystemExit)
+        end
+      end
+
+      context 'when the specified account has a previous migration' do
+        let(:migration) { instance_double(AccountMigration, target_account: target_account, target_account_id: target_account.id) }
+
+        before do
+          allow(Account).to receive(:find_local).with(source_account.username).and_return(source_account)
+          allow(source_account).to receive(:moved_to_account_id).and_return(target_account.id)
+          allow(source_account.migrations).to receive(:last).and_return(migration)
```

Comment:
This is not a showstopper, but I'd create those records in the database rather than stub those fairly low-level calls.